internal/registry/registry_test.go

```diff
@@ -62,6 +71,131 @@ var _ = Describe("ImageExists", func() {
+		})
+
+		It("should fail if it cannot get crane manifest", func() {
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusBadRequest)
+			}))
+			defer server.Close()
+			var u *url.URL
+			u, err = url.Parse(server.URL)
+			Expect(err).ToNot(HaveOccurred())
+			serverHost := strings.Split(u.String(), "/")[2]
+
+			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
+			image := fmt.Sprintf("%s/%s", serverHost, endpoint)
+			_, err = reg.ImageExists(ctx, image, ootov1alpha1.PullOptions{}, nil)
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("failed to get crane manifest from image"))
+		})
+
+		It("should fail if it cannot unmarshal crane manifest", func() {
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// return no error but with an empty manifest (which is not a valid manifest file)
+			}))
+			defer server.Close()
+			var u *url.URL
+			u, err = url.Parse(server.URL)
+			Expect(err).ToNot(HaveOccurred())
+			serverHost := strings.Split(u.String(), "/")[2]
+
+			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
+			image := fmt.Sprintf("%s/%s", serverHost, endpoint)
+			_, err = reg.ImageExists(ctx, image, ootov1alpha1.PullOptions{}, nil)
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("failed to unmarshal crane manifest"))
+		})
+
+		It("should fail if the manifest doesn't contain a 'mediaType' field", func() {
+
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				manifest, err := os.ReadFile("testdata/image_manifest.json")
+				Expect(err).NotTo(HaveOccurred())
+
+				release := unstructured.Unstructured{}
+				err = json.Unmarshal(manifest, &release.Object)
+				Expect(err).NotTo(HaveOccurred())
+				unstructured.RemoveNestedField(release.Object, "mediaType")
+				manifest, err = json.Marshal(release.Object)
+				Expect(err).NotTo(HaveOccurred())
+
+				_, err = w.Write(manifest)
+				Expect(err).NotTo(HaveOccurred())
+			}))
+			defer server.Close()
+			var u *url.URL
+			u, err = url.Parse(server.URL)
+			Expect(err).ToNot(HaveOccurred())
+			serverHost := strings.Split(u.String(), "/")[2]
+
+			endpoint := strings.Join(strings.Split(validImage, "/")[1:], "/")
+			image := fmt.Sprintf("%s/%s", serverHost, endpoint)
+			_, err = reg.ImageExists(ctx, image, ootov1alpha1.PullOptions{}, nil)
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("mediaType is missing from the image"))
+		})
+	})
+
+	It("should not fail if the image doesn't exist", func() {
+
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusNotFound)
+		}))
+		defer server.Close()
+		u, err := url.Parse(server.URL)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
I think it would be worth writing a small `mustParseURL` function that would check for the error. Most tests in this PR would use it.

Reply:
Done.